driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -74,6 +77,74 @@ public MappingManager(Session session, ProtocolVersion protocolVersion) {
         // which nodes might join the cluster later.
         // At least if protocol >=2 we know there won't be any 1.2 nodes ever.
         this.isCassandraV1 = (protocolVersion == ProtocolVersion.V1);
+        session.getCluster().register(new SchemaChangeListenerBase() {
+
+            @Override
+            public void onTableRemoved(TableMetadata table) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (mapper.getTableMetadata().equals(table)) {
+                            LOGGER.error("Table {} has been removed; existing mappers for @Entity annotated {} will not work anymore", table.getName(), mapper.getMappedClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onTableChanged(TableMetadata current, TableMetadata previous) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (previous.equals(mapper.getTableMetadata())) {
+                            LOGGER.warn("Table {} has been altered; existing mappers for @Entity annotated {} might not work properly anymore",
+                                    previous.getName(), mapper.getMappedClass());
+                            it.remove();
```

Comment:
Yes, correct. No need to regenerate it upfront imo – contrary to what happens with udts.